assets/js/modules/analytics/common/account-create.js

```diff
@@ -0,0 +1,195 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useState, Fragment, useCallback } from '@wordpress/element';
+import { getQueryArg } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import Button from '../../../components/button';
+import ProgressBar from '../../../components/progress-bar';
+import { trackEvent } from '../../../util';
+import TimezoneSelect from './timzezone-select';
+import AccountField from './account-field';
+import PropertyField from './property-field';
+import ProfileField from './profile-field';
+
+// import Data from 'googlesitekit-data';
+// const { dispatch, select } = Data;
+
+const AccountCreate = () => {
+	// Uncomment this section after 1311 is merged.
+	/*
+	const isDoingCreateAccount = useSelect(
+		( select ) => {
+			return select( STORE_NAME ).isDoingCreateAccount();
+		},
+		[]
+	);
+	const accountTicketTermsOfServiceURL = useSelect(
+		( select ) => {
+			return select( STORE_NAME ).getAccountTicketTermsOfServiceURL();
+		},
+		[]
+	);
+
+	*/
+	//const { createAccount } = useDispatch( STORE_NAME );
+
+	/*
+	// Redirect if the accountTicketTermsOfServiceURL is set.
+	if ( accountTicketTermsOfServiceURL ) {
+		location = accountTicketTermsOfServiceURL;
+	}
+	*/
+
+	const isDoingCreateAccount = false;
+	const { createAccount } = () => {};
+	const { siteName, siteURL, timezone: tz } = global.googlesitekit.admin;
+	const errorCode = getQueryArg( location.href, 'error_code' );
+
+	// Handle expected provisioning flow error codes.
+	let errorMessage = false;
+	switch ( errorCode ) {
+		case 'user_cancel':
+			errorMessage = __( 'The Terms of Service were not accepted.', 'google-site-kit' );
+			break;
+
+		case 'max_accounts_reached':
+			errorMessage = __( 'The Google Analytics account limit has been reached.', 'google-site-kit' );
+			break;
+
+		case 'backend_error':
+			errorMessage = __( 'Unknown service error.', 'google-site-kit' );
+			break;
+	}
+	const [ error, setError ] = useState( errorMessage );
+
+	const handleSubmit = useCallback( ( accountName, propertyName, profileName, timezone ) => {
+		trackEvent( 'analytics_setup', 'new_account_setup_clicked' );
+		createAccount( {
+			accountName,
+			propertyName,
+			profileName,
+			timezone,
+		} ).then( ( e ) => {
+			const { error: err } = e.payload;
+			if ( err ) {
+				setError( err.message ? err.message : __( 'Unknown error.', 'google-site-kit' ) );
+			}
+		} );
```

Comment:
No need for error handling here based on the above comments, when we start using `getError` it will automatically be displayed. This code here can just become `await createAccount( ... )` (via `useDispatch`) - and afterwards we can do the redirect if the URL (via `useSelect` on `getAccountTicketTermsOfServiceURL`) is not empty.

Reply:
removed

Reply:
not seeing the error handling, let's discuss further.

Reply:
Ok, so I guess `ErrorNotice` is missing, should I go ahead and add that into the component? or should it be on the container component? I'll try placing it here in the component.